### ISSUE TITLE
Provide way to navigate to the next shabad

### ISSUE THREAD

**Issue by sukhpalsingh** [KhalisFoundation/sttm-desktop#11]
Sometimes the shabad searched is showing part of the shabad and need to navigate to the next part of the shabad.

e.g. anand bhiaa meri maae and then need to go to next pauri.

Currently, this will require searching the next pauri and will be really hard to do this 40 times for each anandh sahib pauri.

It will be good to provide arrows or any other ways at the end of shabad to go to the next or previous shabad.

This will help for navigating lots of linked shabads. e.g. japji sahib, asa di vaar, sukhmani sahib.

It will be even better if this happen by short keys also. e.g. shift + right arrow for next shabad and shift + left arrow for previous shabad.

**Comment by tarunsingh5** [comment]
@sukhpalsingh is this something you can help with?

**Comment by ManjotS** [comment]
#27 is a prerequisite for this.

**Comment by ManjotS** [comment]
MySQL Query used by the API for this (needs to be adapted for SQLite):

```(SELECT 'previous' as navigation,PageNo FROM Verse WHERE PageNo=$pageno-1 AND SourceID='$src' LIMIT 1) 
                    UNION
                  (SELECT 'next' as navigation,PageNo FROM Verse WHERE PageNo=$pageno+1 AND SourceID='$src' LIMIT 1);```

**Comment by ManjotS** [comment]
Oops, above is for ang. Here is shabad query:
```
(SELECT 'previous' as navigation,ShabadID FROM Shabad WHERE VerseID=$firstLine-1 LIMIT 1) 
UNION
(SELECT 'next' as navigation,ShabadID FROM Shabad WHERE VerseID=$lastLine+1 LIMIT 1);
```

**Comment by inderpreetsingh** [comment]
```
SELECT
      'previous' as navigation, ShabadID
    FROM
      Shabad
    WHERE
      VerseID='${FirstLine - 1}'
    UNION
    SELECT
      'next' as navigation,
    ShabadID
      FROM
    Shabad
      WHERE
    VerseID='${LastLine + 1}'
```

Surprisingly and annoyingly Brackets do not work like that in SQLite.  Please check if the query I wrote is correct.

**Comment by sukhpalsingh** [comment]
in the logic, make sure that it does not go previous for the first shabad and also does not go next after the last shabad. This will be very minor issue to consider but good to take care of it i guess.

**Comment by sukhpalsingh** [comment]
Anyone working on this?

**Comment by ManjotS** [comment]
I believe @inderpreetsingh has a PR for this

**Comment by sukhpalsingh** [comment]
ok thanks.

**Comment by tarunsingh5** [comment]
fixed in 3.3
